src/probnum/filtsmooth/gaussfiltsmooth/kalman_utils.py

```diff
@@ -0,0 +1,215 @@
+"""Prediction, update, and smoothing implementations for the Kalman filter/smoother."""
+
+import typing
+
+import numpy as np
+
+import probnum.random_variables as pnrv
+
+from .stoppingcriterion import StoppingCriterion
+
+########################################################################################################################
+# Prediction choices
+########################################################################################################################
+
+
+def predict_via_transition(
+    dynamics_model,
+    start,
+    stop,
+    rv,
+    _intermediate_step=None,
+    _linearise_at=None,
+    _diffusion=1.0,
+) -> (pnrv.RandomVariable, typing.Dict):
```

Comment:
these are the only types I am including here. I think it makes sense that all these functions have the same output signature (which I am trying to establish here), but more than that I did not dare as long as we support python 3.6 :)